pkg/kudoctl/cmd/diagnostics/runner_helper.go

```diff
@@ -0,0 +1,167 @@
+package diagnostics
+
+import (
+	"github.com/kudobuilder/kudo/pkg/kudoctl/env"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/util/kudo"
+	"github.com/kudobuilder/kudo/pkg/version"
+)
+
+func diagForInstance(instance string, options *Options, c *kudo.Client, info version.Info, s *env.Settings, p *nonFailingPrinter) error {
+	ir, err := newInstanceResources(instance, options, c, s)
+	if err != nil {
+		p.printError(err, DiagDir, "instance")
+		return err
+	}
+
+	ctx := &processingContext{root: DiagDir, instanceName: instance}
+
+	runner := runnerForInstance(ir, ctx)
+	runner.addObjDump(info, ctx.rootDirectory, "version")
+	runner.addObjDump(s, ctx.rootDirectory, "settings")
+
+	if err := runner.run(p); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func diagForKudoManager(options *Options, c *kudo.Client, p *nonFailingPrinter) error {
+	kr, err := newKudoResources(options, c)
+	if err != nil {
+		return err
+	}
+	ctx := &processingContext{root: KudoDir}
+
+	runner := runnerForKudoManager(kr, ctx)
+
+	if err := runner.run(p); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func runnerForInstance(ir *resourceFuncsConfig, ctx *processingContext) *runner {
+	r := &runner{}
+
+	instance := resourceCollectorGroup{[]resourceCollector{
```

Comment:
I still think that we would be better off with a dedicated high-level collector like `InstanceCollector` than with a generic `resourceCollectorGroup`. But this might be a topic for a later refactoring